src/ng/directive/select.js

```diff
@@ -80,6 +80,9 @@ var SelectController =
 
   // Tell the select control that an option, with the given value, has been added
   self.addOption = function(value, element) {
+    // Skip comment nodes, as they only pollute the `optionsMap`
+    if (element.prop('nodeType') === NODE_TYPE_COMMENT) return;
```

Comment:
AddOption is a private Api so we guarantee that element will always be jq
wrapped. I'm a bit worried about the overhead of calling prop on every
option.
Am 29.01.2016 13:28 schrieb "Georgios Kalpakas" <notifications@github.com>:

> In src/ng/directive/select.js
> <https://github.com/angular/angular.js/pull/13878#discussion_r51255240>:
>
> > @@ -80,6 +80,9 @@ var SelectController =
> >
> >    // Tell the select control that an option, with the given value, has been added
> >    self.addOption = function(value, element) {
> > +    // Skip comment nodes, as they only pollute the `optionsMap`
> > +    if (element.prop('nodeType') === NODE_TYPE_COMMENT) return;
>
> That was my first "reaction", but then I thought: "What if element[0] is
> undefined ?"
> So, instead of element[0] && element[0].nodeType, I went for
> element.prop(...) (I'm not even sure if the element passed to the linking
> function can be empty...)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/angular.js/pull/13878/files#r51255240>.
>